src/scipp/curve_fit.py

```diff
@@ -220,9 +220,11 @@ def _curve_fit(
             )
 
     fda = da.flatten(to='row')
-    mask = irreducible_mask(fda.masks, 'row')
-    if mask is not None:
-        fda = fda[~mask]
+    if len(fda.masks) > 0:
+        _mask = zeros(dims=fda.dims, shape=fda.shape, dtype='bool')
+        for mask in fda.masks.values():
+            _mask |= mask
+        fda = fda[~_mask]
```

Comment:
The `irreducible_mask` function returns `None` if there is a scalar mask. But in that case we want to mask the entire array, that's why I'm not using it here anymore.

Reply:
Could this use something like `_mask = sc.reduce(fda.masks.values()).any()`?

Reply:
No not as far I can tell. That fails if one of the masks is scalar and the other is non-scalar.